subsys/net/lib/lwm2m/lwm2m_rw_json.c

```diff
@@ -605,8 +605,16 @@ int do_write_op_json(struct lwm2m_engine_obj *obj,
 
 			obj_field = lwm2m_get_engine_obj_field(obj,
 							       path->res_id);
+			/*
+			 * if obj_field is not found,
+			 * treat as an optional resource
+			 */
 			if (!obj_field) {
-				return -EINVAL;
+				if (context->operation != LWM2M_OP_CREATE) {
```

Comment:
I suggest to put a "TODO: support BOOTSTRAP WRITE" here
When the “Write” operation targets an Object or an Object Instance, the LwM2M Client MUST ignore optional resources it does not support in the payload. (5.2.3.4)